### PR TITLE
refactor: Code with dependencies_required decorator

### PR DESCRIPTION
**Pull request by Zheng-Lu**
### Refactor Code to Use `dependencies_required` Decorator for Dependency Checks

**Motivation**
To simplify the code and improve maintainability, we should use the `dependencies_required` decorator for dependency checks. This avoids repetitive import and error handling code.

**Solution**
I have refactored the code to replace manual dependency checks with the `dependencies_required` decorator and passed all required tests
image

For example:

**Original Code**
```python
try:
    import neo4j
except ImportError:
    raise ValueError(
        "Could not import neo4j python package. "
        "Please install it with `pip install neo4j`."
    )

# Function that uses neo4j
def some_function():
    pass
```

**Refactored Code**
```python
@dependencies_required('neo4j')
def some_function():
    import neo4j  # This is safe now since the decorator ensures the dependency is installed
    pass
```

### CODE REVIEW

File: camel/toolkits/github_toolkit.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 from camel.functions import OpenAIFunction
 
+from ..utils import dependencies_required
```
Comment (Wendong-Fan) [inline camel/toolkits/github_toolkit.py:22]:
```suggestion
from .utils import dependencies_required
```

**Review by Wendong-Fan — APPROVED** [review]
Thanks @Zheng-Lu! LGTM, just one small fix in below

**Comment by Zheng-Lu** [comment]
I reviewed the failed tests, and they all relate to connection errors caused by the OpenAI API. However, all tests passed successfully in my local environment when I ran `pytest test`. I'm unsure how to resolve these test failures on GitHub, as it seems the OpenAI API is expected to be a developer's local environment variable.

**Review by lightaime — APPROVED** [review]
Looks great. Thanks @Zheng-Lu!

**Comment by Wendong-Fan** [comment]
> I reviewed the failed tests, and they all relate to connection errors caused by the OpenAI API. However, all tests passed successfully in my local environment when I ran `pytest test`. I'm unsure how to resolve these test failures on GitHub, as it seems the OpenAI API is expected to be a developer's local environment variable.

Hey @Zheng-Lu , no worries, it's because your this PR is submitted from your repo which doesn't add api keys required. I have added you into CAMEL org so you can create new branch in CAMEL repo directly next time, thanks for the contribution!

**Pull request merged**